zap/src/main/java/org/parosproxy/paros/network/HttpRequestHeader.java

```diff
@@ -464,9 +463,8 @@ private void parse(boolean isSecure) throws URIException, HttpMalformedHeaderExc
         }
 
         if (mMethod.equalsIgnoreCase(CONNECT)) {
-            parseHostName(sUri);
-            mUri = parseURI(mHostName);
-
+            parseHostNameAndPort(getHeader(HOST));
+            mUri = parseURI(getHeader(HOST));
```

Comment:
The changes done to this class can be reverted per latest changes from upstream.